company-lsp.el

```diff
@@ -366,10 +375,8 @@ See the documentation of `company-backends' for COMMAND and ARG."
                              (company-lsp--candidates-async arg callback))))
          (company-lsp--candidates-sync arg)))
     (sorted t)
-    (no-cache (if (eq company-lsp-cache-candidates 'auto)
-                  (let ((cache (company-lsp--cache-get arg)))
-                    (or (null cache) (company-lsp--cache-incomplete-p cache)))
-                (not company-lsp-cache-candidates)))
+    (no-cache (or (eq company-lsp-cache-candidates 'auto)
```

Comment:
`(not (eq ... t))` might have the same effect.

Reply:
Technically the behavior is different for values other than 'auto, t, or nil. But it's okay since it's undocumented. Done.